### PR TITLE
Special projects "featured content area" widget

### PR DESCRIPTION
**Pull request by joshdarby**
## Changes

This pull request makes the following changes:

- Created the `City Limits Special Projects Featured Content` widget to display the rest of the posts on a series landing page, loosely based off of Largo Recent Posts widget

Desktop:
screencapture-citylimits-test-series-election-watch-2018-2019-11-08-14_03_55

Mobile:
screencapture-citylimits-test-series-election-watch-2018-2019-11-08-14_04_34

## Why

For #73 

## Testing/Questions

Features that this PR affects:

- Special projects series posts being displayed

Questions that need to be answered before merging:

- [x] Is this PR targeting the correct branch in this repository?
- [ ] The widget is named `City Limits Special Projects Featured Content` since that's what it's named in #73, but that feels like it's kind of misleading since this is really just posts that haven't been displayed already. Should we name it something like `City Limits Special Projects Series Archive`?
- I had to redeclare `largo_load_more_posts_choose_partial` since the filter `largo_lmp_template_partial` doesn't pass the `$query_vars` for me to be able to check if the widget is calling LMP. Is there an easier way to do this besides redeclaring this function?
- [ ] Did I miss any edge cases?
- [ ] Does it work as expected?

Steps to test this PR:

1. Find a series with more than ( 7 (7 series widget) + 8 (this widget) = ) 15 posts
2. Add the new `City Limits Seven Series Posts` widget to the landing page and set it to avoid duplicates
3. Add the `City Limits Special Projects Featured Content` widget to a landing page and make sure it's set to avoid duplicate posts
4. Make sure both widgets do not display duplicate posts
5. Find any weird edge cases

### CODE REVIEW

File: wp-content/themes/citylimits/functions.php [hunk 1]
```diff
@@ -606,3 +607,81 @@ function cftl_tax_landing_save_custom_fields( $post_id ){
...
+		$partial = 'search';
+	}
+	// check if this query is for a date, assuming that all date queries have a year.
+	if ( isset($query_vars['year']) && $query_vars['year'] != 0 ) {
+		$partial = 'archive';
+	}
+	// Series landing pages
+	if ( isset($_POST['is_series_landing']) && $_POST['is_series_landing'] == 'true') {
+
+		if( true === $query_vars['is_series_featured_content_widget'] ) {
```
Comment (benlk) [inline wp-content/themes/citylimits/functions.php]:
```
if ( isset( $query_vars['is_series_featured_content_widget'] ) && true === $query_vars['is_series_featured_content_widget'] ) ) 
```

**Comment by benlk** [comment]
> Should we name it something like City Limits Special Projects Series Archive?
Sounds good to me.

> I had to redeclare largo_load_more_posts_choose_partial since the filter largo_lmp_template_partial doesn't pass the $query_vars for me to be able to check if the widget is calling LMP. Is there an easier way to do this besides redeclaring this function?

Aren't they passed to the filter as $post_query->query_vars?

https://github.com/INN/largo/blob/512da701664b329f2f92244bbe54880a6e146431/inc/ajax-functions.php#L157

> Make sure both widgets do not display duplicate posts

If both widgets have checked the box to avoid duplicate posts, then it works.

**Comment by joshdarby** [comment]
Er, yep. I read that wrong because

```
* When building your own filter, you must set the fourth parameter of add_filter to 2:
// 	 *
// 	 *     function your_filter_name($partial, $post_type) {
```

`post_type` got me mixed up. I'll update to use the filter instead of redeclaring the function.

File: wp-content/themes/citylimits/functions.php (hunk 1 above)
Comment (joshdarby) [reply wp-content/themes/citylimits/functions.php]: @benlk https://github.com/INN/umbrella-citylimits/pull/100/commits/7226afc0d4f32bcbac0d0f2eafad9a990c981e76 updates that line and also puts it inside of the filter instead of redeclaring the function.
